astropy/nddata/nddata_base.py

```diff
@@ -90,3 +90,9 @@ def uncertainty(self, value):
                 raise TypeError('Uncertainty must have attribute '
                                 'uncertainty_type whose type is string.')
         self._uncertainty = value
+        #Try setting the parent for the uncertainty. This is currently an
+        #enforced requirement for `StdDevUncertainty` for error propagation.
+        try:
+            self.uncertainty.parent_nddata = self
+        except:
```

Comment:
Please add a specific exception here (e.g. `except AttributeError:` if that is the error generated).